### PR TITLE
Add OS X support

### LINKED ISSUE #2: Add OS X compatibility
- [x] terminal-notifier
- [x] [fswatch](http://stackoverflow.com/questions/1515730/is-there-a-command-like-watch-or-inotifywait-on-the-mac)
- [x] convert

### PR DESCRIPTION
**Pull request by MoOx**
I also exploded the file in 2 parts: init + actual daemon (just for
readability)

Close #2

### CODE REVIEW

File: sharedrop.sh [hunk 1]
```diff
@@ -1,149 +1,75 @@
...
+    hash=$(make_hash "$infile")
+
+    if echo "$infile" | grep -q "." >/dev/null
+    then
+      outfile="$hash.$ext"
+    else
+      outfile="$hash"
+    fi
+
+    # Not sure why I get permission issue
```
Comment (MoOx) [inline sharedrop.sh]: if you got any clue for this...

**Comment by MoOx** [comment]
@Macxim did you try this ?

**Comment by Macxim** [comment]
Not yet. Will definitely try today. I'll let you know.

**Comment by Macxim** [comment]
So far I only tested it locally.

- I had to install terminal-notifier (optional) - `brew install terminal-notifier`
- I had to install fswatch (required) -  `brew install fswatch`
- `/Users/macxim/.local/share/sharedrop/files/list.html: No such file or directory`
- `/Users/macxim/.local/share/sharedrop/files: No such file or directory` I had to create it myself.

I don't know if it helps nor if it's very relevant.
If not, ignore this. I will try to provide a better test ASAP.

```console
2015-05-28T13:09:08 Starting Sharedrop (/Users/macxim/.config/sharedrop, /Users/macxim/.local/share/sharedrop)
2015-05-28T13:09:09 Syncing
2015-05-28T13:09:16 rsync: building file list ... rsync: link_stat "/Users/macxim/.local/share/sharedrop/files/." failed: No such file or directory (2)
2015-05-28T13:09:16 rsync: done
2015-05-28T13:09:16 rsync:
2015-05-28T13:09:16 rsync: sent 33 bytes  received 20 bytes  7.07 bytes/sec
2015-05-28T13:09:16 rsync: total size is 0  speedup is 0.00
2015-05-28T13:09:16 rsync: rsync error: some files could not be transferred (code 23) at /SourceCache/rsync/rsync-45/rsync/main.c(992) [sender=2.6.9]
2015-05-28T13:09:16 Watching /Users/macxim/code/sharedrop
2015-05-28T13:09:23 /Users/macxim/code/sharedrop/Screen Shot 2015-05-28 at 1.05.40 PM.png -> drop:8888:files/mfpfzim8hh3or7h58jxdmk7vdtij3kd.png
2015-05-28T13:09:23 Syncing
2015-05-28T13:09:29 rsync: building file list ... rsync: link_stat "/Users/macxim/.local/share/sharedrop/files/." failed: No such file or directory (2)
2015-05-28T13:09:29 rsync: done
2015-05-28T13:09:29 rsync:
2015-05-28T13:09:29 rsync: sent 33 bytes  received 20 bytes  8.15 bytes/sec
2015-05-28T13:09:29 rsync: total size is 0  speedup is 0.00
2015-05-28T13:09:29 rsync: rsync error: some files could not be transferred (code 23) at /SourceCache/rsync/rsync-45/rsync/main.c(992) [sender=2.6.9]
```

**Comment by madx** [comment]
Guys, I see that you are using this code "in production" so I trust you and merge :dancer: 

I'll try the Linux side when I'm back at work :)

**Pull request merged**